apps/gateway/src/utils/converters.ts

```diff
@@ -170,14 +188,24 @@ export const toOpenAICompatibleMessage = (
   };
 
   if (result.toolCalls && result.toolCalls.length > 0) {
-    message.tool_calls = result.toolCalls.map((toolCall: any) => ({
-      id: toolCall.toolCallId,
-      type: "function" as const,
-      function: {
-        name: toolCall.toolName,
-        arguments: JSON.stringify(toolCall.input),
-      },
-    }));
+    message.tool_calls = result.toolCalls.map((toolCall: any) => {
+      const metadata = {};
+      if (toolCall.providerMetadata) {
```

Comment:
Is providerMetadata the same as providerOptions above or is that different?

Reply:
`providerOptions` is for input to generateText. However, for the results, `providerMetadata` represents the output.